JsonPreprocessor/CJsonPreprocessor.py

```diff
@@ -1428,6 +1445,51 @@ def __removeTokenStr(self, sInput : str) -> str:
                 sInput = sInput.replace(tokenStr.value, '')
         return sInput
 
+    def __preCheckJsonFile(self, sInput, CJSONDecoder):
+        '''
+Checks and handle dynamic path of imported file.
+**Arguments:**
+
+* ``sInput``
+
+  / *Condition*: required / *Type*: str /
+
+* ``CJSONDecoder``
+
+  / *Condition*: required / *Type*: str /
+
+**Returns:**
+
+* ``sInput``
+
+  / *Type*: str /
+        '''
+        try:
+            self.jsonCheck = json.loads(sInput, cls=CJSONDecoder, object_pairs_hook=self.__processImportFiles)
+        except Exception as error:
+            failedJsonDoc = self.__getFailedJsonDoc(error)
+            jsonException = "not defined"
+            if failedJsonDoc is None:
+                jsonException = f"{error}\nIn file: '{self.handlingFile.pop(-1)}'" if len(self.handlingFile)>0 else f"{error}"
+            else:
+                jsonException = f"{error}\nNearby: '{failedJsonDoc}'\nIn file: '{self.handlingFile.pop(-1)}'" if len(self.handlingFile)>0 else \
+                                f"{error}\nNearby: '{failedJsonDoc}'"
+            self.__reset()
+            raise Exception(jsonException)
+        sJson = ''
+        while re.search(r"'\s*\[\s*import\s*\]\s*'", str(self.jsonCheck)):
+            self.__checkDotInParamName(self.jsonCheck)
+            oJson, bNested = self.__updateAndReplaceNestedParam(self.jsonCheck)
+            sJson = json.dumps(oJson)
+            checkDynamicImport = re.search(r'("\s*\[\s*import\s*\]\s*"\s*:\s*"[^"]+")', sJson)
+            if checkDynamicImport is None or '${' in checkDynamicImport[0]:
+                raise Exception('TBD')
+            else:
+                sJson = re.sub(r'"\s*\[\s*import\s*\]\s*":\s*"[${]+[^"]+"', checkDynamicImport[0], sJson)
+            self.__preCheckJsonFile(sJson, CJSONDecoder)
+        self.bDynamicImport = False
```

Comment:
Hi Son,
this won't work most likely.
Once bDynamicImport is set this will affect all later recursions. It will be always set.
The idea to use a instance-variable here is strange for me. It is a matter of each single import if there is a  dynamic import. Why should the whole jsonprerpro instance know this for a specific import.
Please check the design.

Code to demostrate issue:
```
class MyClass:
    def __init__(self):
        # Instanzvariable: hasImport, standardmäßig False
        self.hasImport = False

    def recursive_method(self, count=0):
        # Status von hasImport in der Konsole loggen
        print(f"Rekursionstiefe: {count}, hasImport: {self.hasImport}")

        # Rekursive Bedingung
        if count < 10:
            # Letzte Aktion in der Rekursion
            if count == 4:
                self.hasImport = True
            
            # Rekursiver Aufruf
            self.recursive_method(count + 1)

        self.hasImport = False


# Beispiel zur Verwendung der Klasse
my_instance = MyClass()
my_instance.recursive_method()
```

Thank you,
 Thomas

Reply:
Hello Thomas,

We handle import files in the hook method of ```json.loads()```
```
            oJson = json.loads(sJsonDataUpdated,
                               cls=CJSONDecoder,
                               object_pairs_hook=self.__processImportFiles)
```
then nested parameters will be detected and replaced by the method ```oJson, bNested = self.__updateAndReplaceNestedParam(oJson)```
with this machenism, we could not get values of parameters while ```__processImportFiles```.

I designed to create the method ```__preCheckJsonFile()``` to load and replace parameters in dynamic paths of import files.
While handling JSON configuration file with ```__preCheckJsonFile()```, if it detects dynamic import in JSON config file, the flag ```bDynamicImport``` sets ```True``` until ```__preCheckJsonFile()``` finished.
The flag ```bDynamicImport``` is set to ensure during pre-check phase if there are parameters which are not available (due to these parameters belong to dynamic import files), these exceptions will be bypassed.

After pre-check phase is completed, the flag ```bDynamicImport``` is set ```False```, and all dynamic paths of import files are solved.
Then we call 
```
            oJson = json.loads(sJsonDataUpdated,
                               cls=CJSONDecoder,
                               object_pairs_hook=self.__processImportFiles)
```
and process as normal.

Thank you,
Son

Reply:
Hi Son,
thank you for your answer.
I have still concerns. The nature of recursion is that once your self.bDynamicImport is set, it will be set for all further recursions due to nested imports.
@HolQue : Can you please support with a test-case for this.
Thank you,
 Thomas

Reply:
Test files already available: https://github.com/test-fullautomation/python-jsonpreprocessor/pull/376#issue-2629575318

Tests soon.